sklearn/utils/validation.py

```diff
@@ -366,10 +366,19 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
     dtype_numeric = isinstance(dtype, six.string_types) and dtype == "numeric"
 
     dtype_orig = getattr(array, "dtype", None)
+
     if not hasattr(dtype_orig, 'kind'):
         # not a data type (e.g. a column named dtype in a pandas DataFrame)
         dtype_orig = None
 
+    if dtype_orig is not None and dtype_orig.kind == "c":
+        raise ValueError("Complex data is not supported\n{}\n".format(array))
+    elif isinstance(array, list) or isinstance(array, tuple):
```

Comment:
I'm not sure this is the best way. We might also not catch everything with it (how about dataframes that contain complex data?). I'm not sure this is easy to catch later if ``dtype`` is not None, though. With your addition we're converting into an array twice: here and in 411. It would be great if we could use the conversion at 411, but as I said, it's not entirely clear to me how to do that.